clusterkubevirtadm/cmd/credentials/credentials.go

```diff
@@ -0,0 +1,318 @@
+				Name: opts.Namespace,
+			},
+		}
+		fmt.Printf("Creating namespace %s for cluster %s\n", opts.Namespace, opts.ClusterName)
+		_, err = opts.Client.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to create namespace %s for cluster %s; %w", opts.Namespace, opts.ClusterName, err)
+		}
+		return nil
+	}
+
+	return fmt.Errorf("failed to read namespace %s ; %w", opts.Namespace, err)
+}
+
+func EnsureServiceAccount(ctx context.Context, opts Options) error {
+	_, err := opts.Client.CoreV1().ServiceAccounts(opts.Namespace).Get(ctx, opts.ServiceAccountName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			_, err = createServiceAccount(ctx, opts)
+			if err != nil {
+				return fmt.Errorf("failed to create the service account %s; %w", opts.ServiceAccountName, err)
+			}
+		} else {
+			return fmt.Errorf("can't get the serviceAccount; %w", err)
+		}
+	}
+	return nil
+}
+
+func createServiceAccount(ctx context.Context, opts Options) (*corev1.ServiceAccount, error) {
+	sa := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: opts.ServiceAccountName,
+		},
+	}
+
+	fmt.Println("creating ServiceAccount", opts.ServiceAccountName)
+	_, err := opts.Client.CoreV1().ServiceAccounts(opts.Namespace).Create(ctx, sa, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+
+	// make sure that the related secrets were created for the new serviceAccount. Wait 10 seconds before giving up.
+	ticker := time.Tick(time.Millisecond * 100)
+	for {
+		select {
+		case <-ticker:
+			sa, err = opts.Client.CoreV1().ServiceAccounts(opts.Namespace).Get(ctx, opts.ServiceAccountName, metav1.GetOptions{})
+			if err != nil {
+				return nil, err
+			}
+
+			if len(sa.Secrets) > 0 {
+				return sa, nil
+			}
+
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		}
+	}
+}
+
+func EnsureRole(ctx context.Context, opts Options) (*rbacv1.Role, error) {
+	requiredRole := generateRole(opts)
+	role, err := opts.Client.RbacV1().Roles(opts.Namespace).Get(ctx, requiredRole.Name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			fmt.Println("creating Role", requiredRole.Name)
+			role, err = opts.Client.RbacV1().Roles(opts.Namespace).Create(ctx, requiredRole, metav1.CreateOptions{})
+			if err != nil {
+				return nil, fmt.Errorf("can't create role; %w", err)
+			}
+		} else {
+			return nil, fmt.Errorf("can't read role %s; %w", role.Name, err)
+		}
+	}
+	return role, nil
+}
+
+func generateRole(opts Options) *rbacv1.Role {
+	roleName := fmt.Sprintf("%s-cluster-admin-role", opts.ClusterName)
+
+	return &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleName,
+			Namespace: opts.Namespace,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{rbacv1.APIGroupAll},
+				Resources: []string{rbacv1.ResourceAll},
+				Verbs:     []string{rbacv1.VerbAll},
```

Comment:
i think we can limit this to `get,create,delete` for secrets and `*` for VirtualMachines